packages/clarity-js/src/layout/schema.ts

```diff
@@ -0,0 +1,56 @@
+import { Dimension, Metric, Setting } from "@clarity-types/data";
+import { Constant, JsonLD } from "@clarity-types/layout";
+import * as dimension from "@src/data/dimension";
+import * as metric from "@src/data/metric";
+
+/* JSON+LD (Linked Data) Recursive Parser */
+export function ld(json: any): void {
+    for (let key of Object.keys(json)) {
+        let value = json[key];
+        if (key === JsonLD.Type && typeof value === "string") {
+            value = value.toLowerCase();
+            /* Normalizations */
+            value = value.indexOf(JsonLD.Article) >= 0 || value.indexOf(JsonLD.Posting) >= 0 ? JsonLD.Article : value;
+            switch (value) {
+                case JsonLD.Article:
+                case JsonLD.Recipe:
+                    dimension.log(Dimension.SchemaType, json[key]);
+                    break;
+                case JsonLD.Product:
+                    dimension.log(Dimension.SchemaType, json[key]);
+                    dimension.log(Dimension.ProductName, json[JsonLD.Name]);
+                    dimension.log(Dimension.ProductSku, json[JsonLD.Sku]);
+                    if (json[JsonLD.Brand]) { dimension.log(Dimension.ProductBrand, json[JsonLD.Brand][JsonLD.Name]); }
+                    break;
+                case JsonLD.Rating:
+                    if (json[JsonLD.RatingValue]) {
+                        let bestRating = num(json[JsonLD.BestRating], Setting.BestRatingValue);
+                        let rating = (parseFloat(json[JsonLD.RatingValue]) * Setting.RatingScale) / bestRating;
+                        metric.max(Metric.RatingValue, rating);
+                    }
+                    metric.max(Metric.RatingCount, num(json[JsonLD.RatingCount]));
+                    metric.max(Metric.ReviewCount, num(json[JsonLD.ReviewCount]));
+                    break;
+                case JsonLD.Author:
+                    dimension.log(Dimension.AuthorName, json[JsonLD.Name]);
+                    break;
+                case JsonLD.Offer:
+                    dimension.log(Dimension.ProductAvailability, json[JsonLD.Availability]);
+                    dimension.log(Dimension.ProductCondition, json[JsonLD.ItemCondition]);
+                    dimension.log(Dimension.Currency, json[JsonLD.PriceCurrency]);
+                    dimension.log(Dimension.ProductSku, json[JsonLD.Sku]);
+                    metric.max(Metric.ProductPrice, num(json[JsonLD.Price]));
+                    break;
+                case JsonLD.Brand:
```

Comment:
These rely on typing only but not the "path" to the property. Example: brand.name is the property path. If the type of the brand property is not "Brand" then we wouldn't pick up the attribute. This would complicate this a bit but it would be ideal if the code kept the "current" path around and let us use that for the matching as well as the type name. 

Ignore if you think this is too much.

Reply:
Yeah let's come back to it later.